### PR TITLE
libexif: add tests and examples

### PR DESCRIPTION
**Pull request by neheb**
Plus some cleanups.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
very strange

library() requires vs_module_defs and shared_library does not.

**Comment by neheb** [comment]
@jpakkane this is ready.

File: subprojects/packagefiles/libexif/meson.build [hunk 1]
```diff
@@ -2,89 +2,72 @@ project('exif', 'c', license: 'LGPL-2.1', version: '0.6.24')
...
-    'libexif/olympus/mnote-olympus-tag.c',
-    'libexif/fuji/exif-mnote-data-fuji.c',
-    'libexif/fuji/mnote-fuji-entry.c',
-    'libexif/fuji/mnote-fuji-tag.c'
-]
+defs = custom_target(
+  input: 'libexif/libexif.sym',
+  output: 'libexif.def',
+  command: ['def.py', '@INPUT@', '@OUTPUT@'],
+)
```
Comment (eli-schwartz) [inline subprojects/packagefiles/libexif/meson.build:60]: This should really be used on all platforms, not just Windows... bit annoying to create it and then test if it's valid as a `-Wl,.....` flag everywhere though.

Comment (neheb) [reply subprojects/packagefiles/libexif/meson.build:60]: Annoying indeed.

**Comment by neheb** [comment]
ping @xclaesse

**Comment by neheb** [comment]
@xclaesse rebased.

File: subprojects/packagefiles/libexif/meson.build [hunk 2]
```diff
@@ -2,89 +2,73 @@ project('exif', 'c', license: 'LGPL-2.1', version: '0.6.24')
...
-    include_directories : '.',
-    soversion: 0, version: '0.6.22'
-    )
+exif = library(
+  'exif',
+  sources,
+  vs_module_defs: defs,
+  install: true,
+  dependencies: [m_dep, intl_dep],
+  include_directories: '.',
```
Comment (jpakkane) [inline subprojects/packagefiles/libexif/meson.build:69]: Not needed.

Comment (neheb) [reply subprojects/packagefiles/libexif/meson.build]: Fixed.

**Pull request merged**